From 07d19399f25eda971c60b48242ae448438d91bb7 Mon Sep 17 00:00:00 2001
From: Sultanxda <sultanxda@gmail.com>
Date: Sun, 6 Sep 2015 02:16:18 -0700
Subject: [PATCH 041/211] SnapdragonCamera: Reset camera state after taking
 picture

mCameraState gets stuck to SNAPSHOT_IN_PROGRESS after taking a picture using autofocus/not ZSL, thus breaking camera capture.

Properly reset the camera state after taking a picture to fix this.

Change-Id: Ie4bd629d9e1db1ed3e5b5f80c5b02fd000d88c8e
---
 src/com/android/camera/PhotoModule.java | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/com/android/camera/PhotoModule.java b/src/com/android/camera/PhotoModule.java
index 3854eec68..da49753b1 100644
--- a/src/com/android/camera/PhotoModule.java
+++ b/src/com/android/camera/PhotoModule.java
@@ -2968,6 +2968,11 @@ public class PhotoModule
         mErrorCallback.setActivity(mActivity);
         mCameraDevice.setErrorCallback(mErrorCallback);
 
+        // Reset camera state after taking a picture
+        if (mCameraState != PREVIEW_STOPPED && mCameraState != INIT) {
+            setCameraState(IDLE);
+        }
+
         // Preview needs to be stopped when changing resolution
         if (mRestartPreview && mCameraState != PREVIEW_STOPPED && mCameraState != INIT) {
             stopPreview();
-- 
2.17.1

